From 5f92e4ef7c797e92f391d7e0323492be39a70613 Mon Sep 17 00:00:00 2001
From: Ciprian Marian Costea <ciprianmarian.costea@nxp.com>
Date: Sun, 8 Aug 2021 16:31:04 +0300
Subject: [PATCH 09/19] s32-gen1: serdes: Correctly set phy id

commit 5f92e4ef7c797e92f391d7e0323492be39a70613 from
https://source.codeaurora.org/external/autobsps32/linux

In 'serdes_xlate', the phy id was set to either PCIE0 or PCIE1,
which is wrong because each phy should correspond to a single lane.

Upstream-Status: Pending

Issue: ALB-6998
Signed-off-by: Ciprian Marian Costea <ciprianmarian.costea@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/phy/freescale/phy-fsl-s32gen1-serdes.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/phy/freescale/phy-fsl-s32gen1-serdes.c b/drivers/phy/freescale/phy-fsl-s32gen1-serdes.c
index c7edf7c3aa40..a8a75438e550 100644
--- a/drivers/phy/freescale/phy-fsl-s32gen1-serdes.c
+++ b/drivers/phy/freescale/phy-fsl-s32gen1-serdes.c
@@ -597,7 +597,7 @@ static struct phy *serdes_xlate(struct device *dev,
 		return ERR_PTR(ret);
 
 	phy = serdes->phys[lane_id];
-	phy->id = instance;
+	phy->id = lane_id;
 	phy->attrs.mode = mode;
 
 	return phy;
-- 
2.17.1

